webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -775,28 +772,17 @@ trait PromiseGeneration_BuildNodeContext {
   def systemVarService: SystemVariableService
 
   // we should get the context to replace the value (PureResult[String] to String)
-  def parseJValue(value: JValue, context: InterpolationContext): JValue = {
-    def rec(v: JValue): JValue = v match {
-      case JObject(l) => JObject(l.map { field => field.copy(value = rec(field.value)) })
-      case JArray(l)  => JArray(l.map(rec))
-      case JString(s) =>
-        // What to do if the compilation fails ?
-        //  1. Keep the value by default and log error ?
-        //  2. Return a RudderError ?
-        //  3. Other ?
-        (for {
-          v <- interpolatedValueCompiler.compile(s).chainError(s"Error when looking for interpolation variable '${s}' in node property")
-          s <- v(context)
-        } yield {
-          s
-        }) match {
-          case Left(err) =>
-            // Not sure about what to log if we don't want to send and error
-            PolicyGenerationLogger.error(s"Interpolation variable in node property fails, keeping value ${s}, cause: ${err.fullMsg}")
-            JString(s)
-          case Right(interpolatedString) => JString(interpolatedString)
-        }
-      case x          => x
+  def parseJValue(value: JValue, context: InterpolationContext): IOResult[JValue] = {
```

Comment:
this was the main culprit: we do want to let the user know that their is a problem. And so it trickle down all over the place.